test/test_autograd.py

```diff
@@ -4527,6 +4527,33 @@ def test(inp, inp_dtype, out_dtype):
         test(inp, torch.float, torch.double)
         test(inp, torch.double, torch.float)
 
+    def test_nan_to_num(self):
+        a = torch.randn(3, 3, 3, 3)
+        with torch.no_grad():
+            a[torch.randn_like(a) < 0.2] = float('nan')
```

Comment:
If you use randn here, <0.2 will be most of the Tensor no? Maybe you want something smaller?

Reply:
Right. Was supposed to be `torch.rand_like(a) < 0.2`